src/psyclone/psyGen.py

```diff
@@ -4616,9 +4616,10 @@ def begin_string(self):
         table = self.scope.symbol_table
         readers = set()
         writers = set()
-        for var in var_accesses.all_vars:
+        for signature in var_accesses.all_signatures:
```

Comment:
Yes, but we then need three lines of list comprehension to convert the signatures to strings :) That was the reason why I stopped using Signature. I was only not sure if the set subtraction would work as expected (since there are likely different Signature instances with the same value) ... but they do work (makes sense, sets are based on dicts, and Signatures define an appropriate hash function).

But I agree, using signatures is nicer, so I've added code to convert the lists of signatures to lists of strings.

Reply:
Hmm - I added a comment (in the 'file changed' tab, it doesn't show up here :( See details elsewhere I guess.

Reply:
Looks good actually. Thanks for changing.